ControlRoomApplication/ControlRoomApplication/Controllers/PLCCommunication/PLCDrivers/ProductionPLCDriver.cs

```diff
@@ -35,6 +37,16 @@ public class ProductionPLCDriver : AbstractPLCDriver
         private MCUManager MCU;
         private RadioTelescopeTypeEnum telescopeType = RadioTelescopeTypeEnum.NONE;
 
+
+        // Previous snow dump azimuth -- we need to keep track of this in order to add 45 degrees each time we dump
+        private double previousSnowDumpAzimuth;
+
+        // Snow dump timer
+        private static System.Timers.Timer snowDumpTimer;
+
+        // Weather data for thermal calibration
+        private AbstractWeatherStation weatherStation;
```

Comment:
Upon closer investigation, you won't actually have to declare a `watherStation` object. I'll explain why in the next comments: